### DISCUSSION TITLE
Pico 2 W Firmware

### DISCUSSION THREAD

**Discussion by snphysic** [micropython/micropython#16363] [Core Development]
I recently purchased an RP2 Pico 2 W for my project, intending to use its Bluetooth functionality. Unfortunately, I’ve encountered an issue:

The Pico W firmware does not work with the Pico 2 W.
The Pico 2 firmware, while functional, does not support Bluetooth.
Could you kindly let me know if there are plans to release a firmware version for the Pico 2 W that supports Bluetooth? I am eager to use this feature and would appreciate any updates on the timeline.

Thank you for your time and assistance.

**Comment by scruss** [comment]
firmware is here: https://downloads.raspberrypi.com/micropython/mp_firmware_unofficial_latest.uf2
from this discussion: [support for raspberrypi pico 2 W · micropython · Discussion #16314](https://github.com/orgs/micropython/discussions/16314)

**Reply by snphysic** [reply]
I really appreciate it.

**Comment by mattytrentini** [comment]
PR #16313 contains the RPi Pico 2 W board definition; it's currently under review. Please watch it if you're interested - and help test if possible!
